include/seqan/platform/platform_gcc.h

```diff
@@ -39,9 +39,20 @@
  * @signature #define PLATFORM_GCC
  */
 
+// GNU COMPATIBLE
 #ifndef PLATFORM_GCC
 #define PLATFORM_GCC
 
+#if !defined(__clang__) && !defined(__ICC) && !defined(PLATFORM_GNU)
```

Comment:
That check is already done before including the header platform_gcc.h 
So inside the header we just need to make sure its nothing else that real gcc.